server/sonar-server/src/main/java/org/sonar/server/component/ws/ProjectMeasuresQueryFactory.java

```diff
@@ -65,6 +66,11 @@ private static void processCriterion(FilterParser.Criterion criterion, ProjectMe
       return;
     }
 
+    if (QUERY_KEY.equalsIgnoreCase(key)) {
+      processQuery(criterion, query);
+      return;
+    }
+
     String value = criterion.getValue();
     checkArgument(value != null, "Value cannot be null for '%s'", key);
```

Comment:
remark: I'm not sure about the key naming. Good for operator and values but key?